src/main/java/org/openstreetmap/atlas/checks/validation/linear/lines/WaterWayCheck.java

```diff
@@ -0,0 +1,529 @@
+                    object.getOsmIdentifier());
+            if (flag == null)
+            {
+                flag = createFlag(object, instruction, Collections.singletonList(last));
+            }
+            else
+            {
+                flag.addObject(object, last, instruction);
+            }
+
+        }
+        final LineItem crossed = intersectsAnotherWaterWay(atlas, line);
+        if (crossed != null)
+        {
+            final Iterator<Location> intersections = crossed.asPolyLine()
+                    .intersections(line.asPolyLine()).iterator();
+            if (intersections.hasNext())
+            {
+                final String instruction = this.getLocalizedInstruction(
+                        FALLBACK_INSTRUCTIONS.indexOf(CROSSES_WATERWAY), object.getOsmIdentifier(),
+                        crossed.getOsmIdentifier());
+                if (flag == null)
+                {
+                    flag = createFlag(Sets.hashSet(object, crossed), instruction,
+                            Arrays.asList(intersections.next()));
+                }
+                else
+                {
+                    flag.addObject(crossed, intersections.next(), instruction);
+                }
+            }
+        }
+        if (flag == null)
+        {
+            return Optional.empty();
+        }
+        super.markAsFlagged(object.getOsmIdentifier());
+        return Optional.of(flag);
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * Create a flag for an object that goes uphill
+     *
+     * @param flag
+     *            The flag to create/modify. May be {@code null}.
+     * @param object
+     *            The object to flag
+     * @param first
+     *            The first node of the way
+     * @return The new CheckFlag (if the passed flag was {@code null}) or the modified CheckFlag.
+     */
+    private CheckFlag createUphillFlag(final CheckFlag flag, final AtlasObject object,
+            final Location first)
+    {
+        CheckFlag returnFlag = flag;
+        final String instruction = this.getLocalizedInstruction(
+                FALLBACK_INSTRUCTIONS.indexOf(GOES_UPHILL), object.getOsmIdentifier(),
+                this.elevationUtils.getResolution(first).asMeters());
+        if (returnFlag == null)
+        {
+            return createFlag(object, instruction);
+        }
+        returnFlag.addInstruction(instruction);
+        return returnFlag;
+    }
+
+    /**
+     * Check if the waterway ends in a sink (i.e., a location that can reasonably expected to have
+     * no outflow).
+     *
+     * @param line
+     *            The waterway item to check
+     * @return {@code true} if the waterway ends in a sink.
+     */
+    private boolean doesWaterwayEndInSink(final LineItem line)
+    {
+        return doesLineEndOnWaterway(line) || doesLineEndInSink(line) || doesLineEndInOcean(line);
```

Comment:
Missing `this` keyword.

Reply:
That wasn't part of the style guide. (The guide being `./gradlew spotlessCheck`). Should look into updating that to force methods to have `this` as a qualifier?

Reply:
I think it would be good to add it to spotlessCheck. @taylorsmock